lib/urldata.h

```diff
@@ -60,13 +60,18 @@
  * protocol fields in the protocol handler.
  */
 #define CURLPROTO_WS     (1<<30)
-#define CURLPROTO_WSS    (1LL<<31)
-
-/* This type should be bumped to a curl_off_t once we need bit 32 or higher */
-typedef unsigned int curl_prot_t;
+#define CURLPROTO_WSS    ((curl_prot_t)1<<31)
 #else
 #define CURLPROTO_WS 0
 #define CURLPROTO_WSS 0
+#endif
+
+/* This should be undefined once we need bit 32 or higher */
+#define PROTO_TYPE_SMALL
+
+#ifndef PROTO_TYPE_SMALL
+typedef curl_off_t curl_prot_t;
```

Comment:
wouldn't this check be easier and make more sense if reversed?

Like with `#ifdef PROTY_TYPE_BIG` ... 

So that we have a define we *set* when we cross the 32 boundary and need to use a larger type? Especially since that new protocol will require a 64 bit data type, so going beyond 32 needs to be conditional (at least until we require a 64 bit type).

Reply:
I don't think so. I started to implement it as `PROTO_TYPE_LARGE`, but I noted this is easier to undefine a symbol than to define it and I reversed the logic.
- `#undef` can be applied on a undefined symbol: it's just ignored.
- `#define` **might** redefine an existing symbol only if the value does not change. But this is implementation-dependent.

When several new protocols (!) will be introduced, they will all need to act on this, so this will be easier to simply `#undef` the symbol than to redefine it for each protocol.

I know this is future, but we have to prepare for it :-)

Reply:
Ok, fair enough. We can always change it later if we think of improvements down the line.